### ISSUE TITLE
How to make it work with gunicorn

### ISSUE THREAD

**Issue by prabin525** [jaydenwindle/graphene-subscriptions#21]
I am using graphene-subscription in one of my project. Everything is fine but the problem is whenever i use gunicorn for my server, it doesn't work. I tried running the daphne on 8001 port but whatever I try, I get the following issue

`{
  "error": "Could not connect to websocket endpoint ws://127.0.0.1:8001. Please check if the endpoint url is correct."
}`

**Comment by jaydenwindle** [comment]
Hi @prabin525!

Because `graphene-subscriptions` uses Django Channels behind the scenes (which uses ASGI instead of WSGI), you won't be able to access the subscriptions endpoint when running your app with a WSGI server like gunicorn.

What path did you define for your subscriptions endpoint in `routing.py`? Right now it looks like you're trying to connect to `ws://127.0.0.1:8001/`. If you followed the instructions from the readme when setting up your app, your subscriptions endpoint might be listening on `ws://127.0.0.1:8001/graphql` instead.

Can you try connecting to `ws://127.0.0.1:8001/graphql` and see if it works? If not I'm happy to dive a little deeper and help you debug things :)

**Comment by prabin525** [comment]
Thank you @jaydenwindle for your reply. I got it to work.  There is another problem that I am facing. The subscriptions work fine with playground but as our frontend application is react based, we use apollo-graphql and whenever I try to run this subscription it gives the following error:
`
"GraphQL error: Subscriptions are not allowed. You will need to either use the subscribe function or pass allow_subscriptions=True"
`

The same error is given in the graphiql and Insomnia client. What might be the way to fix this?

**Comment by jaydenwindle** [comment]
@prabin525 It sounds like your React app might be trying to send subscription requests via Apollo's HTTP link to your GraphQL endpoint (`http://<your_api>/graphql` instead of `ws://<your_api>/graphql`). Have you followed the [Client Setup](https://www.apollographql.com/docs/react/data/subscriptions/#client-setup) instructions for subscriptions in the Apollo documentation?

Currently `graphene-subscriptions` doesn't play well with `graphene-django`'s default GraphiQL setup (which doesn't support subscriptions). I'd guess that Insomnia doesn't connect via Websocket to your subscription endpoint by default either. I'd recommend testing subscriptions with GraphQL Playground since it will auto-connect to the subscription Websocket endpoint if it's the same as the HTTP endpoint.

**Comment by prabin525** [comment]
@jaydenwindle Thank you for your suggestions and I made it to work perfectly. I have used daphne to serve both HTTP and WS requests. The problem is that the websockets connect only when the debug of django is TRUE (i.e in development mode only) but in the production mode the websocekts doesn't connect. There is no error thrown too. I also changed the djangoo channels Inchannel layer to redis. But nothing seems to work. What may be the way forward to remove this error.
